pkg/projectfile/projectfile_darwin_test.go

```diff
@@ -0,0 +1,43 @@
+package projectfile
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/environment"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestYamlMerge(t *testing.T) {
```

Comment:
It's great that you wrote a test for this, but this is too reliant on implicit behaviors and external factors.

I think instead we should take the merge code and put it into a dedicated function eg. `merge(projectStruct1, projectStruct2)` and test that. That way we're not reliant on IO nor any platform specific logic.